### ISSUE TITLE
Dimensions with multiple datatypes

### ISSUE THREAD

**Issue by herrstucki** [zazuko/rdf-cube-view-query#45]
In [this case](https://s.zazuko.com/2NEeT), the dimension with path `<https://environment.ld.admin.ch/foen/ubd0066_t3/wert>` should have a `xsd:decimal` datatype but it's missing from the generated shape. I assume this is because of "cube:Undefined" values.

Note that zazuko/cube-creator#633 fixed one part of this issue only, the min/max properties are now correct.

Generated shape:
```
_:genid-5f4c31b08327482faca54aec279620a3-14738E10580C19031E26F4927DD18C56 <http://schema.org/name> "Concentration en mg/kg MS"@fr , "Konzentration in mg/kg TS"@de ;
   <http://www.w3.org/ns/shacl#path> <https://environment.ld.admin.ch/foen/ubd0066_t3/wert> ;
   <http://www.w3.org/ns/shacl#nodeKind> <http://www.w3.org/ns/shacl#Literal> ;
   <http://www.w3.org/ns/shacl#minCount> 1 ;
   <http://www.w3.org/ns/shacl#maxCount> 1 ;
   <http://www.w3.org/ns/shacl#minInclusive> 0.2098 ;
   <http://www.w3.org/ns/shacl#maxInclusive> 58.5113 ;
   <http://qudt.org/schema/qudt/unit> <http://qudt.org/vocab/unit/MilliGM-PER-KiloGM> ;
   <http://qudt.org/schema/qudt/scaleType> <http://qudt.org/schema/qudt/RatioScale> ;
```

**Comment by herrstucki** [comment]
@bergos what would make the most sense here? Add a `dimension.datatypes` property? What would `dimension.datatype` then be?

- Just picking the first may result in the datatype being `cube:Undefined` (depending on the list order)
- Is there any guarantee that only one other datatype than undefined is generated?

For reference, this is what I've currently implemented in my code to get all non-undefined datatypes:
```js
const dataTypes = [
      ...(dim.out(ns.sh.or).list() ?? dim.out(ns.sh.or).toArray()),
    ].flatMap((item) => {
      return item
        .out(ns.sh.datatype)
        .filter((d) => !ns.cube.Undefined.equals(d.term)).terms;
    })
```

I'm ignoring undefined in my code and hoping that there's only one other datatype used 😅

**Comment by bergos** [comment]
fixed in #52. See the comment of the PR for more details.
